query-node/mappings/src/storage/metadata.ts

```diff
@@ -66,6 +73,52 @@ async function processNodeLocationMetadata(
   return nodeLocation
 }
 
+export function processNodeOperationalStatusMetadata(
+  actorContext: 'lead' | 'worker',
+  current: typeof NodeOperationalStatus | undefined,
+  meta: INodeOperationalStatusMetadata
+): typeof NodeOperationalStatus {
+  const isCurrentForced =
+    current &&
+    (current instanceof NodeOperationalStatusNoService ||
+      current instanceof NodeOperationalStatusNoServiceFrom ||
+      current instanceof NodeOperationalStatusNoServiceDuring) &&
+    current.forced
+
+  // if current state is forced by lead, then prevent worker from unilaterally reversing.
+  if (isCurrentForced && actorContext === 'worker') {
+    return current
+  }
+
+  // set node state to NoService
+  if (meta.status === NodeOperationalStatusMetadata.OperationalStatus.NO_SERVICE) {
+    if (meta.noServiceFrom && meta.noServiceTo) {
+      const status = new NodeOperationalStatusNoServiceDuring()
+      status.from = new Date(meta.noServiceFrom)
+      status.to = new Date(meta.noServiceTo)
```

Comment:
Are those raw dates validated anywhere? If not, we probably should do that

Reply:
Addressed this in subsquid PR https://github.com/Joystream/storage-squid/pull/19